src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -426,100 +432,103 @@ public void acceptRecord(final SAMRecord record) {
+            ///////////////////////////////////////////////////////////////////
+            if (record.getReadUnmappedFlag()) return;
 
             // Prefetch the list of target and bait overlaps here as they're needed multiple times.
-            final Collection<Interval> targets;
-            final Collection<Interval> probes;
-
-            if (!rec.getReadUnmappedFlag()) {
-                final Interval read = new Interval(rec.getReferenceName(), rec.getAlignmentStart(), rec.getAlignmentEnd());
-                targets = targetDetector.getOverlaps(read);
-                probes  = probeDetector.getOverlaps(read);
-            }
-            else {
-                targets = null;
-                probes = null;
-            }
-
-            // get the number of bases above the minimum base quality
-            for (final byte baseQuality : rec.getBaseQualities()) {
-                if (baseQuality < minimumBaseQuality) metrics.PCT_EXC_BASEQ++;
-            }
-
-            if (!rec.getSupplementaryAlignmentFlag()) {
-                // And now calculate the values we need for HS_LIBRARY_SIZE
-                if (rec.getReadPairedFlag() && rec.getFirstOfPairFlag() && !rec.getReadUnmappedFlag() && !rec.getMateUnmappedFlag()) {
-                    if (probes != null && !probes.isEmpty()) {
+            final Interval read = new Interval(record.getReferenceName(), record.getAlignmentStart(), record.getAlignmentEnd());
+            final Collection<Interval> targets = targetDetector.getOverlaps(read);
+            final Collection<Interval> probes  = probeDetector.getOverlaps(read);
+
+            // Calculate the values we need for HS_LIBRARY_SIZE
+            if (!record.getSupplementaryAlignmentFlag()) {
+                if (record.getReadPairedFlag() && record.getFirstOfPairFlag() && !record.getReadUnmappedFlag() && !record.getMateUnmappedFlag()) {
+                    if (!probes.isEmpty()) {
                         ++this.metrics.PF_SELECTED_PAIRS;
-                        if (!rec.getDuplicateReadFlag()) ++this.metrics.PF_SELECTED_UNIQUE_PAIRS;
+                        if (!record.getDuplicateReadFlag()) ++this.metrics.PF_SELECTED_UNIQUE_PAIRS;
                     }
                 }
             }
 
-            // Check for reads that are marked as duplicates or that didn't align uniquely
-            if (rec.getDuplicateReadFlag() || rec.getReadUnmappedFlag() || rec.getMappingQuality() == 0) {
+            ///////////////////////////////////////////////////////////////////
+            // Duplicate reads can be totally ignored beyond this point
+            ///////////////////////////////////////////////////////////////////
+            if (record.getDuplicateReadFlag()) {
+                this.metrics.PCT_EXC_DUPE += basesAlignedInRecord;
                 return;
             }
 
-            final boolean mappedInPair = rec.getReadPairedFlag() && !rec.getMateUnmappedFlag() && !rec.getSupplementaryAlignmentFlag();
```

Comment:
didn't matter here to check `rec.getReadUnmappedFlag()` since we returned above.